common/validators/base_validators.py

```diff
@@ -0,0 +1,151 @@
+"""
+Copyright 2023 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+from datetime import datetime, timezone
+from decimal import Decimal
+from enum import Enum
+from typing import Optional
+
+from validataclass.dataclasses import Default, ValidataclassMixin, validataclass
+from validataclass.validators import (
+    BooleanValidator,
+    DateTimeValidator,
+    EnumValidator,
+    IntegerValidator,
+    ListValidator,
+    Noneable,
+    NumericValidator,
+    StringValidator,
+)
+
+from util import LotData, LotInfo
+
+
+class ParkingSiteTypeInput(Enum):
+    ON_STREET = 'ON_STREET'
+    OFF_STREET_PARKING_GROUND = 'OFF_STREET_PARKING_GROUND'
+    UNDERGROUND = 'UNDERGROUND'
+    CAR_PARK = 'CAR_PARK'
+    OTHER = 'OTHER'
+
+    def to_lot_info_type(self) -> str:
+        return {
+            self.ON_STREET: 'street',
+            self.OFF_STREET_PARKING_GROUND: 'lot',
+            self.UNDERGROUND: 'underground',
+            self.CAR_PARK: 'garage',
+            self.OTHER: 'unknown',
+        }.get(self, 'unknown')
```

Comment:
I would argue that "Parkdeck" is almost impossible to differ from car park, and also DATEX2 has no support for that, but that might be something we could add. How would you translate it? PARKING_DECK? ROOFTOP_PARKING? And how should we differentiate this from an actual garage, which also has parking decks and often rooftop parking?

Bus is a vehicle type and no parking type, so I would keep the data model clean in a new version and remove it.

Reply:
Hi there, 
the `bus` type came from a lot in Dresden (i think). They are actually listing the capacity of a parking lot for public busses, which has no real use for people searching a lot but i found it interesting to record nevertheless. It's not listed in the API.

Reply:
@hbruch and I already discussed about getting the vehicle type in it, because bike parking is also a topic, but this would be an own property in my opinion. Although this easily gets extremely complex, because you can also mix vehicle type and specific properties, eg "one bike and one car parking place with charging abilities, additionally, 10 car, 2 bus and 20 bike parkings without". To make a good compromise there is not really easy, but to do that keeping the data model as clean as possible is somehow a good idea in my opinion.